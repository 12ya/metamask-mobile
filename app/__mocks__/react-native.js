import { Linking, DeviceEventEmitter } from 'react-native';

Linking.addEventListener = jest.fn();
Linking.removeEventListener = jest.fn((event, callback) => {
  // Simulate the behavior of removeEventListener
  if (typeof callback === 'function') {
    callback();
  }
});
Linking.openURL = jest.fn();
Linking.canOpenURL = jest.fn();
Linking.getInitialURL = jest.fn();

DeviceEventEmitter.addListener = jest.fn();
DeviceEventEmitter.removeListener = jest.fn((event, callback) => {
  // Simulate the behavior of removeListener
  if (typeof callback === 'function') {
    callback();
  }
});

export default { Linking, DeviceEventEmitter };
