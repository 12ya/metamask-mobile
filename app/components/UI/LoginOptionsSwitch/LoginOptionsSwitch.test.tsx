import React from 'react';
import { render, screen } from '@testing-library/react-native';
import LoginOptionsSwitch from './LoginOptionsSwitch';
import { BIOMETRY_TYPE } from 'react-native-keychain';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

describe('LoginOptionsSwitch', () => {
  const mockStore = configureMockStore();
  // eslint-disable-next-line @typescript-eslint/no-empty-function
  const handleUpdate = (_biometricsEnabled: boolean) => {};

  it('should render correctly with biometric option', () => {
    const store = mockStore({});
    const { toJSON } = render(
      <Provider store={store}>
        <LoginOptionsSwitch
          shouldRenderBiometricOption={BIOMETRY_TYPE.FACE}
          biometryChoiceState
          onUpdateBiometryChoice={handleUpdate}
          onUpdateRememberMe={handleUpdate}
        />
      </Provider>,
    );
    expect(toJSON()).toMatchSnapshot();
    expect(screen.getByText('Face ID')).toBeTruthy();
    expect(screen.getByText('Remember me')).toBeTruthy();
  });

  it('should not render biometric option when shouldRenderBiometricOption is null', () => {
    const store = mockStore({});
    render(
      <Provider store={store}>
        <LoginOptionsSwitch
          onUpdateBiometryChoice={handleUpdate}
          onUpdateRememberMe={handleUpdate}
          biometryChoiceState
          shouldRenderBiometricOption={null}
        />
      </Provider>,
    );
    expect(screen.queryByText('Face ID')).toBeNull();
    expect(screen.getByText('Remember me')).toBeTruthy();
  });
});