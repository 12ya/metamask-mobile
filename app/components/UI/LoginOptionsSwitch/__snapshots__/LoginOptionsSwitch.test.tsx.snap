// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginOptionsSwitch should not render biometric option when shouldRenderBiometricOption is null 1`] = `null`;

exports[`LoginOptionsSwitch should render correctly with biometric option 1`] = `
<View
  style={
    {
      "marginBottom": 30,
      "marginTop": 20,
      "position": "relative",
    }
  }
>
  <Text
    style={
      {
        "color": "#141618",
        "flex": 1,
        "fontFamily": "EuclidCircularB-Regular",
        "fontSize": 16,
        "fontWeight": "400",
      }
    }
  >
    [missing "en.biometrics.enable_face" translation]
  </Text>
  <RCTSwitch
    accessibilityRole="switch"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onTintColor="#0376c9"
    style={
      [
        {
          "height": 31,
          "width": 51,
        },
        [
          {
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "backgroundColor": "#bbc0c566",
            "borderRadius": 16,
          },
        ],
      ]
    }
    testID="login-with-biometrics-switch"
    thumbTintColor="#ffffff"
    tintColor="#bbc0c566"
    value={true}
  />
</View>
`;
